graphql-async/test/async_test.ml

```
@@ -16,14 +16,26 @@ let test_query schema ctx query expected =
     match Graphql_parser.parse query with
     | Error err -> failwith err
     | Ok doc ->
-      Graphql_async.Schema.execute schema ctx doc >>| fun result ->
-      let result' = match result with
-      | Ok (`Response data) -> data
-      | Error err -> err
-      in
-      Alcotest.check yojson "invalid execution result" expected result'
+      Graphql_async.Schema.execute schema ctx doc >>= (function
+      | Ok (`Response data) -> Async_kernel.return data
+      | Ok (`Stream stream) ->
+          Async_kernel.Pipe.to_list stream >>| fun lst ->
+            `List (
+              List.fold_right (fun x acc -> match x with
+                | Ok data -> data :: acc
+                | _ -> acc) lst [])
```

Suggestion:

```ocaml
Async_kernel.Pipe.to_list stream >>| fun lst ->
`List Core_kernel.(List.filter_map lst ~f:Result.ok)
```

---

... though I don't know if it's good we're ignoring the errors in the tests?